types/block.go

```diff
@@ -305,6 +140,23 @@ func (b *Block) MakePartSet(partSize uint32) *PartSet {
 	return NewPartSetFromData(bz, partSize)
 }
 
+// TODO(Wondertan): Aim for Block to be complete without calling RowSet
+func (b *Block) RowSet(ctx context.Context, adder format.NodeAdder) (*RowSet, error) {
+	shares, dataLen := b.ComputeShares()
+	eds, err := ipld.PutData(ctx, shares, adder)
```

Comment:
Computing the rowset now also writes to IPFS? Is that intentional?
I think the least invasive approach would be:

1. compute rowset where needed without touching storage yet
2. use rowset (like previously partsset) during consensus and in the other places
3. as a first step only store the block where store.SaveBlock is called (during consensus this is during finalize commit)

Later we can re-consider 3. and optimize to store and provide the data earlier.